lib/libvarnish/vcli_serve.c

```diff
@@ -428,6 +440,12 @@ VCLS_AddFd(struct VCLS *cs, int fdi, int fdo, cls_cb_f *closefunc, void *priv)
 	cfd->cli->vlu = VLU_New(cfd, cls_vlu, *cs->maxlen);
 	cfd->cli->sb = VSB_new_auto();
 	cfd->cli->limit = cs->limit;
+	if (cs->ws_sz > 0) {
+		cfd->cli->ws = malloc(sizeof(*cfd->cli->ws));
+		assert(cfd->cli->ws != NULL);
+		WS_Init(cfd->cli->ws, "cli",
+		    malloc(cs->ws_sz), cs->ws_sz);
```

Comment:
If we embed the ws everywhere we should maybe do it as part of WS_Init (or introduce a non-breaking WS_Embed function).